### PR TITLE
Instant Search: Optimize bundle size and polyfill delivery

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
* Removes usage of `lodash/get` (-0.46kB gzipped from bundle).
* Replaces `cache` with `tiny-lru` (-4.34kB gzipped from bundle).
* Adds usage-based polyfilling via Babel's `preset-env` (+0.32kB gzipped to bundle).
* Adds custom polyfill approach for IE 11. 
  * Polyfill consists of two parts: a loader script and a payload script.
  * The loader script will conditionally adds a script tag for the payload script if IE11 is detected.
  * The payload consists of [`core-js`](https://github.com/zloirock/core-js#commonjs-api) and [`regenerator-runtime/runtime`](https://github.com/facebook/regenerator/tree/master/packages/regenerator-runtime).
  * Both the loader and payload are excluded from our bundle size calculations.

In summary, this PR reduces the bundle size from 53.27 kB gzipped to 48.79 kB gzipped.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack Search site.
* Run a production build.
* Ensure that the search interface works as expected.
* Ensure that the `ie11-polyfill-loader` script is loaded in all environments.
* In IE11, ensure that the `ie11-polyfill-payload` is loaded via `ie11-polyfill-loader`.

#### Proposed changelog entry for your changes:
* Improved Jetpack Search support for older browsers, such as Internet Explorer 11.

### CODE REVIEW

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

**Comment by jeherve** [comment]
I'll merge this for now so we can ship it in 9.4. I'll let you handle its WordPress.com counterpart.

**Comment by jsnmoon** [comment]
Shipped in r220057-wpcom with additional changes now being backported to Jetpack via #18554.
